src/nebari/schema.py

```diff
@@ -17,7 +17,7 @@
 email_regex = "^[^ @]+@[^ @]+\\.[^ @]+$"
 email_pydantic = pydantic.constr(regex=email_regex)
 
-github_url_regex = "^(https://)?github.com/([^/]+)/([^/]+)/?$"
+github_url_regex = "^(https://)?(github.com|gitlab.com)/([^/]+)/([^/]+)/?$"
```

Comment:
How do you feel about writing something like:

```python
git_repo_url_regex = "^(https://)?(git)(hub|lab)(.com)/([^/]+)/([^/]+)/?$"
```

- to remove repetition of common string `git` and `.com`
- also it's no more just github :)

Reply:
Thanks, Amit! That is indeed much nicer. I just made a slight modification at the end to avoid allowing urls with a trailing slash that do not have a repo name. For example:
```
https://gitlab.com/user/
```

This was not being caught before.